akka-projection-cassandra/src/main/scala/akka/projection/cassandra/javadsl/CassandraProjection.scala

```diff
@@ -36,15 +34,11 @@ object CassandraProjection {
   def atLeastOnce[Offset, Envelope](
       projectionId: ProjectionId,
       sourceProvider: SourceProvider[Offset, Envelope],
-      saveOffsetAfterEnvelopes: Int,
-      saveOffsetAfterDuration: Duration,
-      handler: Handler[Envelope]): Projection[Envelope] =
-    scaladsl.CassandraProjection.atLeastOnce(
-      projectionId,
-      new SourceProviderAdapter(sourceProvider),
-      saveOffsetAfterEnvelopes,
-      saveOffsetAfterDuration.asScala,
-      new HandlerAdapter(handler))
+      handler: Handler[Envelope]): AtLeastOnceCassandraProjection[Envelope] =
+    // FIXME refactor SourceProviderAdapter into CassandraProjectionProviderAdapter?
+    scaladsl.CassandraProjection
+      .atLeastOnce(projectionId, new SourceProviderAdapter(sourceProvider), new HandlerAdapter(handler))
+      .asInstanceOf[AtLeastOnceCassandraProjection[Envelope]]
```

Comment:
or we could `new CassandraProjectionImpl` directly here instead of going via scaladsl

Reply:
:+1: